docs/designs/update-strategy.md

```diff
@@ -0,0 +1,661 @@
+
+A user with excess cluster capacity wants to maintain full service capacity during updates. They would like to:
+
+- Surge additional aggregated worker replicas before deleting old ones
+- Keep existing replicas running to meet SLAs while new replicas are created
+- Only delete old replicas after new ones are ready
+
+However, the current default strategy (`maxUnavailable=1, maxSurge=0`) requires deleting an old replica before creating a new one. While service availability is maintained (when replica count > 1), this reduces capacity during the update window. With excess capacity available, users should be able to surge replicas to maintain full capacity throughout the update.
+
+**Scenario 2: Faster Updates with Multiple Replicas**
+
+A user with 4 replicas of aggregated workers wants to speed up the update process. They would like to:
+
+- Update 2 replicas simultaneously (`maxUnavailable=2`) instead of one at a time
+- Reduce overall update time while still maintaining service availability
+- Balance update speed with acceptable capacity reduction during the update window
+
+The current default strategy only allows `maxUnavailable=1`, forcing sequential updates even when the deployment can tolerate multiple replicas being unavailable simultaneously.
+
+## Goals
+
+- **Enable application version incompatible updates** through PodCliqueSet-level replica recreation strategy, allowing entire PCS replicas to be torn down atomically to prevent cross-version communication issues
+
+- **Enable surge capability** at the PCSG level for multinode deployments and PC level for singlenode inference, allowing new replicas to be created before old ones are deleted to maintain full capacity during updates
+
+- **Enable faster rollouts** by allowing multiple replicas to be recreated simultaneously at the PCSG level for multinode deployments and PC level for singlenode inference, configurable via `maxUnavailable` settings
+
+## Non-Goals
+
+- Partition-based rolling updates or canary deployments - These are advanced features that can be considered in future phases
+
+- Automatic strategy selection based on version compatibility detection - Users must explicitly configure update strategies
+
+- Automatic selection of surge based on capacity - Users are expected to be aware of their cluster capacity and should explicitly configure surge settings. Grove does not have visibility into available cluster capacity at update time and cannot automatically determine if surge is feasible
+
+## Proposal
+
+This proposal expands Grove's default RollingUpdate strategy to support replica recreation at the PodCliqueSet level for version incompatible upgrades, as well as `maxUnavailable` and `maxSurge` configuration at the PCS level. Additionally, it introduces `maxUnavailable` and `maxSurge` concepts at the standalone PodClique and PodCliqueScalingGroup levels to enable faster rollouts and capacity maintenance during compatible upgrades.
```

Comment:
Aside from added/work complexity, why not maxUnavailable and maxSurge at the PCS level as well?